src/osd/scheduler/mClockScheduler.cc

```diff
@@ -139,45 +137,148 @@ void mClockScheduler::set_osd_mclock_cost_per_io()
   }
 }
 
-void mClockScheduler::set_client_allocations()
+void mClockScheduler::set_mclock_profile()
 {
-  // Set profile specific client capacity allocations
-  if (mclock_profile == "balanced") {
-    double capacity = std::round(0.5 * max_osd_capacity);
-    client_allocs[op_scheduler_class::client] = capacity;
-    client_allocs[op_scheduler_class::background_recovery] = capacity;
-  } else if (mclock_profile == "high_recovery_ops") {
-    client_allocs[op_scheduler_class::client] =
-      std::round(0.25 * max_osd_capacity);
-    client_allocs[op_scheduler_class::background_recovery] =
-      std::round(0.75 * max_osd_capacity);
-  } else if (mclock_profile == "high_client_ops") {
-    client_allocs[op_scheduler_class::client] =
-      std::round(0.75 * max_osd_capacity);
-    client_allocs[op_scheduler_class::background_recovery] =
-      std::round(0.25 * max_osd_capacity);
-  } else {
-    ceph_assert("Invalid mclock profile" == 0);
-    return;
-  }
+  mclock_profile = cct->_conf.get_val<std::string>("osd_mclock_profile");
 }
 
-double mClockScheduler::get_client_allocation(op_type_t op_type)
+std::string mClockScheduler::get_mclock_profile()
 {
-  double default_allocation = 1.0;
+  return mclock_profile;
+}
 
-  switch (op_type) {
-  case op_type_t::client_op:
-    return client_allocs[op_scheduler_class::client];
-  case op_type_t::bg_recovery:
-    return client_allocs[op_scheduler_class::background_recovery];
-  default:
-    // TODO for other op types.
-    return default_allocation;
-  }
+void mClockScheduler::set_balanced_profile_allocations()
+{
+  // Client Allocation:
+  //   reservation: 40% | weight: 1 | limit: 100% |
+  // Background Recovery Allocation:
+  //   reservation: 40% | weight: 1 | limit: 150% |
+  // Background Best Effort Allocation:
+  //   reservation: 20% | weight: 1 | limit: max |
```

Comment:
I think we'd want a limit on best effort for all these profiles

Reply:
Sure, I wasn't sure about the limits for best effort clients. For now I have provided the following limits,
balanced - 100% of max_osd_capacity
high_client_ops  and high_recovery_ops - 50% of max_osd_capacity.

I think a new profile catering to the best effort clients may be warranted. At this point I am not sure how much allocation
is necessary for clients like scrubbing, snaptrim and pg_delete and how to test them.